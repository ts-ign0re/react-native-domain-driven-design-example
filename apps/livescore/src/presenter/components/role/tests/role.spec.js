import React from 'react';
import renderer from 'react-test-renderer';

import { Role } from '../index';
import { Text, View } from 'react-native';

describe('enabled roles', () => {
  beforeAll(() => {
    Role.setConfig({ display: true });
  });

  test('Roles should be rendered', () => {
    const component = renderer.create(
      <Role name="parent">
        <View>
          <Role name=":child">
            <span>
              <Role name="absolute">
                <Text>Some inner text</Text>
              </Role>
            </span>
          </Role>
        </View>
      </Role>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should pass style to the child', () => {
    const component = renderer.create(
      <Role name="parent" style={{ flex: 1 }}>
        <View />
      </Role>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should merge classes', () => {
    const component = renderer.create(
      <Role name="parent" style={{ flex: 2 }}>
        <View style={{ flex: 1 }} />
      </Role>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should pass unknown props to the child', () => {
    const component = renderer.create(
      <Role name="parent" unknownProp="value">
        <View />
      </Role>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('disabled roles', () => {
  beforeAll(() => {
    Role.setConfig({ display: false });
  });

  test("Roles shouldn't be rendered", () => {
    const component = renderer.create(
      <Role name="parent">
        <View>
          <Role name=":child">
            <span>
              <Role name="absolute">
                <Text>Some text</Text>
              </Role>
            </span>
          </Role>
        </View>
      </Role>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should pass style to the child', () => {
    const component = renderer.create(
      <Role name="parent" style={{ flex: 2 }}>
        <View />
      </Role>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should merge classes', () => {
    const component = renderer.create(
      <Role name="parent" style={{ flex: 2 }}>
        <View style={{ flex: 1 }} />
      </Role>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should pass unknown props to the child', () => {
    const component = renderer.create(
      <Role name="parent" unknownProp="value">
        <View />
      </Role>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
