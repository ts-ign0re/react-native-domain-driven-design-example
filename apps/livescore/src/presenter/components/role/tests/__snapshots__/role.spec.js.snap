// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disabled roles Roles shouldn't be rendered 1`] = `
<View
  style={
    [
      undefined,
      undefined,
    ]
  }
>
  <span
    style={
      [
        undefined,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          undefined,
          undefined,
        ]
      }
    >
      Some text
    </Text>
  </span>
</View>
`;

exports[`disabled roles Should merge classes 1`] = `
<View
  style={
    [
      {
        "flex": 2,
      },
      {
        "flex": 1,
      },
    ]
  }
/>
`;

exports[`disabled roles Should pass style to the child 1`] = `
<View
  style={
    [
      {
        "flex": 2,
      },
      undefined,
    ]
  }
/>
`;

exports[`disabled roles Should pass unknown props to the child 1`] = `
<View
  style={
    [
      undefined,
      undefined,
    ]
  }
  unknownProp="value"
/>
`;

exports[`enabled roles Roles should be rendered 1`] = `
<View
  accessibilityLabel="parent"
  style={
    [
      undefined,
      undefined,
    ]
  }
  testID="parent"
>
  <span
    accessibilityLabel="parent:child"
    style={
      [
        undefined,
        undefined,
      ]
    }
    testID="parent:child"
  >
    <Text
      accessibilityLabel="absolute"
      style={
        [
          undefined,
          undefined,
        ]
      }
      testID="absolute"
    >
      Some inner text
    </Text>
  </span>
</View>
`;

exports[`enabled roles Should merge classes 1`] = `
<View
  accessibilityLabel="parent"
  style={
    [
      {
        "flex": 2,
      },
      {
        "flex": 1,
      },
    ]
  }
  testID="parent"
/>
`;

exports[`enabled roles Should pass style to the child 1`] = `
<View
  accessibilityLabel="parent"
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
  testID="parent"
/>
`;

exports[`enabled roles Should pass unknown props to the child 1`] = `
<View
  accessibilityLabel="parent"
  style={
    [
      undefined,
      undefined,
    ]
  }
  testID="parent"
  unknownProp="value"
/>
`;
